provisioning/device/src/client.ts

```diff
@@ -2,74 +2,53 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 'use strict';
-import * as Provisioning from './interfaces';
-import { ClientStateMachine } from './client_state_machine';
+import { ProvisioningTransportHandlersX509, ProvisioningTransportHandlersBase, ProvisioningDeviceConfiguration } from './interfaces';
+import { errors } from 'azure-iot-common';
 import * as dbg from 'debug';
 const debug = dbg('azure-device-provisioning:device-client');
 
-// tslint:disable-next-line:no-var-requires
-const packageJson = require('../package.json');
+export class ProvisioningDeviceClient {
```

Comment:
>ProvisioningDeviceClient [](start = 13, length = 24)

if you look at the PR i have for TPM I believe this is equivalent to the TpmProvisioning class I have.

initially we started with the idea that the device client would be agnostic of TPM vs x509 and protocol: we'd create it with a security client, and a transport. based on the security client we get, we should determine what type of ProvisioningClient we're using (TpmProvisioning or X509Provisioning: these would not be public objects). then each of those would have their own "flow" state machine and would call into the transport and security client as needed. Of course there's an additional complexity/separation that I don't have on the TPM side that is, you have a polling state machine. That polling state machine should be hidden within the transport layers themselves - the polling mechanism is a transport-specific thing, even if all transports share it (for now), it really is a transport-layer responsibility.

This looks to me like the X509Provisioning class, not the ProvisioningDeviceClient